static/app/views/discover/table/tableActions.tsx

```diff
@@ -159,14 +163,22 @@ function FeatureWrapper(props: FeatureWrapperProps) {
 }
 
 function TableActions(props: Props) {
+  const {tableData, queryDataset, supportsInvestigationRule} = props;
   const location = useLocation();
+  const organization = useOrganization();
   const cursor = location?.query?.cursor;
   const cursorOffset = parseCursor(cursor)?.offset ?? 0;
-  const numSamples = props.tableData?.data?.length ?? null;
+  const numSamples = tableData?.data?.length ?? null;
   const totalNumSamples = numSamples === null ? null : numSamples + cursorOffset;
+
+  const isNotTransactionsView =
+    hasDatasetSelector(organization) &&
+    queryDataset &&
+    queryDataset !== SavedQueryDatasets.TRANSACTIONS;
+
   return (
     <Fragment>
-      {props.supportsInvestigationRule && (
+      {supportsInvestigationRule && !isNotTransactionsView && (
```

Comment:
nit: Up to you, but I found this is a little confusing to read, because the `!` and variable (`isNotTransactionsView`) forms a double negation. Should we simplify this a bit, or break up the condition a little to be more readable? Maybe something like this:
```
const datasetUnselected = !hasDatasetSelector(organization) || !queryDataset;
const isTransactionsView = queryDataset === SavedQueryDatasets.TRANSACTIONS;
...
{supportsInvestigationRule && (datasetUnselected || isTransactionsView) && (
```